### PR TITLE
set it so we use common dockerfile

### PR DESCRIPTION
**Pull request by allandegnan**
This replaced the GRO Dockerfile with the generic hod homeoffice Dockerfile.  Uses ONBUILD to standardise our repos so there's not very much login in this itself.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,7 @@
-FROM vaijab/nodejs:0.12.7
-
-RUN dnf install -y -q git
+FROM quay.io/ukhomeofficedigital/nodejs:v3.0.0
```
Comment (JoeChapman) [inline Dockerfile]: `v3.0.0`?

Comment (allandegnan) [inline Dockerfile]: that's the semver version of this repo, as opposed to the node version.  happy to make it just the same as the node version though

**Comment by JoeChapman** [comment]
@allandegnan this dockerfile appears to be missing some important steps, like `npm install`?

**Comment by JoeChapman** [comment]
@allandegnan please can you add a description to this PR?

**Comment by JoeChapman** [comment]
`docker build .`

```
Cleaning up list of fastest mirrors
npm ERR! fetch failed https://registry.npmjs.org/mocha-junit-reporter/-/mocha-junit-reporter-1.11.0.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 404
npm ERR! fetch failed https://registry.npmjs.org/mocha-junit-reporter/-/mocha-junit-reporter-1.11.0.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 404
npm ERR! fetch failed https://registry.npmjs.org/mocha-junit-reporter/-/mocha-junit-reporter-1.11.0.tgz
npm ERR! Linux 4.1.19-boot2docker
npm ERR! argv "/opt/nodejs/bin/node" "/opt/nodejs/bin/npm" "--production=false" "install" "--no-optional"
npm ERR! node v4.4.2
npm ERR! npm  v2.15.0

npm ERR! fetch failed with status code 404
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /app/npm-debug.log
The command '/bin/sh -c yum clean all &&             yum update -y &&             yum install -y git &&             yum clean all &&             rpm --rebuilddb &&             rm -rf node_modules &&             npm --production=false install --no-optional &&             NODE_ENV=development npm test &&             npm prune --production &&             chown -R nodejs:nodejs .' returned a non-zero code: 1
```

**Comment by allandegnan** [comment]
That's not really related to this PR, just means that npmjs registry is no longer serving that link.

**Comment by allandegnan** [comment]
@JoeChapman This now builds without error.  Please test and confirm.

```
docker pull quay.io/ukhomeofficedigital/nodejs:v4.4.2
docker build -t test .
docker run -e NODE_ENV=docker -p 8080:8080 test 
```

**Comment by JoeChapman** [comment]
redis server is running, but .....

```
GRO@0.0.1 (gh:feature/gfm-19/dsp-deployment*) $ docker run -e NODE_ENV=docker -p 8080:8080 test
starting the service

> GRO@0.0.1 start /app
> node .

{"level":"info","message":"App listening on port 8080","timestamp":"2016-04-18T08:36:44.642Z"}
{"date":"Mon Apr 18 2016 08:36:44 GMT+0000 (UTC)","process":{"pid":18,"uid":997,"gid":996,"cwd":"/app","execPath":"/opt/nodejs/bin/node","version":"v4.4.2","argv":["/opt/nodejs/bin/node","/app"],"memoryUsage":{"rss":80244736,"heapTotal":57937504,"heapUsed":31109712}},"os":{"loadavg":[0.05419921875,0.455078125,0.29931640625],"uptime":333},"trace":[{"column":24,"file":"/app/node_modules/redis/index.js","function":"RedisClient.on_error","line":196,"method":"on_error","native":false},{"column":14,"file":"/app/node_modules/redis/index.js","function":"","line":106,"method":null,"native":false},{"column":13,"file":"events.js","function":"emitOne","line":77,"method":null,"native":false},{"column":7,"file":"events.js","function":"Socket.emit","line":169,"method":"emit","native":false},{"column":8,"file":"net.js","function":"emitErrorNT","line":1253,"method":null,"native":false},{"column":9,"file":"node.js","function":"nextTickCallbackWith2Args","line":442,"method":null,"native":false},{"column":17,"file":"node.js","function":"process._tickCallback","line":356,"method":"_tickCallback","native":false}],"stack":["Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379","    at RedisClient.on_error (/app/node_modules/redis/index.js:196:24)","    at Socket.<anonymous> (/app/node_modules/redis/index.js:106:14)","    at emitOne (events.js:77:13)","    at Socket.emit (events.js:169:7)","    at emitErrorNT (net.js:1253:8)","    at nextTickCallbackWith2Args (node.js:442:9)","    at process._tickCallback (node.js:356:17)"],"level":"error","message":"uncaughtException: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379","timestamp":"2016-04-18T08:36:44.655Z"}

npm ERR! Linux 4.1.19-boot2docker
npm ERR! argv "/opt/nodejs/bin/node" "/opt/nodejs/bin/npm" "start"
npm ERR! node v4.4.2
npm ERR! npm  v2.15.0
npm ERR! code ELIFECYCLE
npm ERR! GRO@0.0.1 start: `node .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the GRO@0.0.1 start script 'node .'.
npm ERR! This is most likely a problem with the GRO package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs GRO
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls GRO
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /app/npm-debug.log
```

**Pull request merged**